lib/modules/datasource/hermit/index.ts

```diff
@@ -0,0 +1,153 @@
+import { cache } from '../../../util/cache/package/decorator';
+import { GithubHttp } from '../../../util/http/github';
+import { id } from '../../versioning/hermit';
+import { Datasource } from '../datasource';
+import { GithubReleasesDatasource } from '../github-releases';
+import { getApiBaseUrl } from '../github-releases/common';
+import type {
+  GithubRelease,
+  GithubReleaseAsset,
+} from '../github-releases/types';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import type { HermitSearchResult } from './types';
+
+const datasource = 'hermit';
+
+/**
+ * Hermit Datasource searches a given package from the specified `hermit-pakcages`
+ * repository. It expects the search manifest to come from an asset `index.json` from
+ * a release named index. Any error fetching, parsing and finding the package from the given
+ * registryUrl will be thrown
+ */
+export class HermitDatasource extends Datasource {
+  static readonly id = datasource;
+
+  override readonly customRegistrySupport = true;
+
+  override readonly registryStrategy = 'first';
+
+  override readonly defaultVersioning = id;
+
+  override readonly defaultRegistryUrls = [
+    'https://github.com/cashapp/hermit-packages',
+  ];
+
+  githubHttp: GithubHttp;
+
+  constructor() {
+    super(HermitDatasource.id);
+    this.githubHttp = new GithubHttp(GithubReleasesDatasource.id);
+  }
+
+  @cache({
+    namespace: `datasource-${datasource}-package`,
+    key: ({ registryUrl, packageName }: GetReleasesConfig) =>
+      `${registryUrl}-${packageName}`,
+  })
+  async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    logger.trace(`HermitDataSource.getReleases()`);
+
+    if (!registryUrl) {
+      throw new Error('registryUrl must be supplied');
+    }
+
+    const res = await this.getHermitRelease(packageName, registryUrl);
+    const sourceUrl = res.Repository;
+
+    return {
+      sourceUrl,
+      releases: [
+        ...res.Versions.map((v) => ({
+          version: v,
+          sourceUrl,
+        })),
+        ...res.Channels.map((v) => ({
+          version: v,
+          sourceUrl,
+        })),
+      ],
+    };
+  }
+
+  /**
+   * getHermitSearchManifest fetch the index.json from release
+   * named index, parses it and returned the parsed JSON result
+   */
+  @cache({
+    namespace: `datasource-${datasource}-search-manifest`,
+    key: ({ registryUrl }: GetReleasesConfig) => registryUrl ?? '',
+  })
+  async getHermitSearchManifest(
+    registryUrl: string
+  ): Promise<HermitSearchResult[]> {
+    if (!registryUrl.startsWith('https://github.com')) {
+      throw new Error(`Only Github registryUrl is supported`);
+    }
+
+    const u = new URL(registryUrl);
+    const host = u.host;
+    const path = u.pathname;
+
+    const parts = path.split('/').filter((p) => p !== '');
+
+    if (parts.length < 2) {
+      throw new Error(`can't find owner & repo in the url`);
```

Comment:
see other comment. do we really need custom registry support?

Reply:
We do need custom registry support. We in Square have our own `hermit-packages` containing internal packages that won't be available in the open sourced `cashapp/hermit-packages`.